modules/storage/storage.go

```diff
@@ -0,0 +1,287 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zosv2/modules"
+	"github.com/threefoldtech/zosv2/modules/storage/filesystem"
+)
+
+const (
+	cacheTarget = "/var/cache"
+	cacheLabel  = "zos-cache"
+	cacheSize   = 20 * 1024 * 1024 * 1024 // 20GB
+)
+
+var (
+	diskBase = map[modules.RaidProfile]int{
+		modules.Single: 1,
+		modules.Raid1:  2,
+		modules.Raid10: 4,
+	}
+)
+
+type storageModule struct {
+	volumes []filesystem.Pool
+	devices filesystem.DeviceManager
+}
+
+// New create a new storage module service
+func New() modules.StorageModule {
+	return &storageModule{
+		volumes: []filesystem.Pool{},
+		devices: filesystem.DefaultDeviceManager(),
+	}
+}
+
+/**
+Initialize, must be called at least onetime each boot.
+What Initialize will do is the following:
+ - Try to mount prepared pools (if they are not mounted already)
+ - Scan free devices, apply the policy.
+ - If new pools were created, the pool is going to be mounted automatically
+**/
+func (s *storageModule) Initialize(policy modules.StoragePolicy) error {
```

Comment:
As we agreed on the chat. I really think it's better if we split this into 2 parts
- The initialize part, only called on the module start, and not exposed on zbus which mounts and loads the exiting pools and ensuring the cache is mounted.
- Another method for the creation of new pools, given the policy.

Reply:
Indeed initialize should not be expose over zbus if the module calls it automatically when it starts